freud/box.pyx

```diff
@@ -494,6 +495,37 @@ cdef class Box:
 
         return np.asarray(distances)
 
+    def crop(self, points):
+        R"""Returns mask corresponding to particle membership in a box.
+
+        Args:
+            points (:math:`\left(N, 3\right)` :class:`numpy.ndarray`):
+                Array of points.
+
+        Returns:
+            :math:`\left(N, \right)` :class:`numpy.ndarray`:
+                Array of booleans, where `True` corresponds to points within the box,
```

Comment:
I think @klywang's chosen language (inside/outside) is unambiguous here. The box has finite volume and defined boundaries, even though periodic boundary conditions are intended to behave as though the box is infinite. @klywang @tommy-waltmann I have some background info below on topics related to this that may be of interest to you and future developers.

freud assumes periodic boundary conditions in virtually all methods. This is an assumption that we often want to relax, in cases like @klywang has here, creating a virtual box that is a subset of a larger box and can be used to "crop." We have long-standing issues (#209, #573) from users in our group and external users where explicit support for non-periodic boxes would be helpful. We have flags for tracking box periodicity that are almost never used internally, and this is not the desired behavior.

To Tommy's question: whether a set of input points "obeys periodic boundary conditions" is not always well-defined, and it depends largely on what users are trying to achieve. Consider cases like polymers where polymer beads may be wrapped such that the molecule is contiguous and its center of mass is inside the periodic box, even though some of its components are outside the box. In such a case, does the molecule "obey" periodic boundary conditions?

Another subtlety here that @klywang got right, but I wanted to echo the motivation -- we try to use "points" instead of "particles" in all docstrings (wherever appropriate) because many tools in the freud library are fairly generic and have some general applicability outside the space of nanoparticle self-assembly. Particles could be atoms (much smaller scale), but they could also be points in a geometry sense, or even data points in a 2D/3D vector space that obeys periodic boundary conditions (I have used freud for such a thing). According to a post on our forum from a while back, we have even had users leverage the library for analyzing crowd dynamics.

Reply:
@bdice In regards to your "points" vs "particles" discussion -- is there also some ambiguity associated with "particles" when working with polygons/polyhedra? My understanding is that some analysis uses center of mass, while others use particle vertices for analysis.

Reply:
All of this is good background, but I don't think it necessarily addresses Tommy's core concern that users accustomed to periodic boundaries may be confused by this. The core problem is that freud's convention for what is "inside" is documented in the tutorials I wrote [here](https://freud.readthedocs.io/en/latest/gettingstarted/examples/module_intros/box.Box.html?highlight=origin#Using-boxes) and [here](https://freud.readthedocs.io/en/latest/gettingstarted/tutorial/periodic.html?highlight=origin#periodic-boundary-conditions), but not in the Box class itself. The latter link is referenced in the Box docstring, but you have to follow the link and it's still kind of hidden in a note at the bottom.

I suggest we move that note to the docstring for the Box class so that we clearly indicate the convention that boxes go from -L/2 to L/2, then reference that here (something like "where inside is defined by the conventions defined by :class:`~.Box`, ignoring periodicity"). Then we have something unambiguous that we can point to when users are confused as to why (0.9, 0, 0) is not inside a box of size (1, 1, 1).

Reply:
@klywang points vs. particles: I think most users put the center of mass at (0, 0, 0) in the particle frame. We don't have any methods in freud that use polytope vertices.